src/main/webapp/app/shared/components/atoms/number-input/number-input.component.ts

```diff
@@ -0,0 +1,100 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit, computed, input, output, signal } from '@angular/core';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+import { AbstractControl, FormControl, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { InputNumberModule } from 'primeng/inputnumber';
+
+@Component({
+  selector: 'jhi-number-input',
+  templateUrl: './number-input.component.html',
+  styleUrl: './number-input.component.scss',
+  standalone: true,
+  imports: [CommonModule, FormsModule, FontAwesomeModule, InputNumberModule, ReactiveFormsModule, InputNumberModule],
+})
+export class NumberInputComponent implements OnInit {
+  control = input<AbstractControl | undefined>(undefined);
+  disabled = input<boolean>(false);
+  icon = input<string | undefined>(undefined);
+  label = input<string | undefined>(undefined);
+  placeholder = input<string | undefined>(undefined);
+  required = input<boolean>(false);
+  error = input<boolean>(false);
+  model = input<number | undefined>(undefined);
+  modelChange = output<number>();
+  labelPosition = input<'top' | 'left'>('top');
+  width = input<string>('100%');
+  inputId = input<string | undefined>(undefined);
+
+  // Optional min and max values
+  min = input<number | undefined>(undefined);
+  max = input<number | undefined>(undefined);
+
+  // Optional min and max fraction digits
+  minFractionDigits = input<number | undefined>(undefined);
+  maxFractionDigits = input<number | undefined>(undefined);
+
+  readonly formValidityVersion = signal(0);
+
+  readonly inputState = computed(() => {
+    this.formValidityVersion();
+
+    if (!this.isTouched()) return 'untouched';
+    if (this.formControl?.invalid === true) return 'invalid';
+    return 'valid';
+  });
+
+  // State tracking
+  isTouched = signal(false);
+  isFocused = signal(false);
+
+  get formControl(): FormControl | undefined {
+    const ctrl = this.control();
+    return ctrl ? (ctrl as FormControl) : undefined;
+  }
+
+  ngOnInit(): void {
```

Comment:
please avoid ngoninit and use a more reactive approach

for example:
```suggestion
  constructor() {
    effect((onCleanup) => {
      const ctrl = this.control() as FormControl;
      this.formControl.set(ctrl);
      if (!ctrl) return;
      const sub = ctrl.statusChanges.subscribe(() => {
        this.formValidityVersion.update(v => v + 1);
      });
      onCleanup(() => sub.unsubscribe());
    });
  }
```